src/idl_gen_go.cpp

```diff
@@ -103,10 +103,12 @@ class GoGenerator : public BaseGenerator {
     bool needs_imports = false;
     for (auto it = parser_.enums_.vec.begin(); it != parser_.enums_.vec.end();
          ++it) {
-      tracked_imported_namespaces_.clear();
-      needs_math_import_ = false;
-      needs_bytes_import_ = false;
-      needs_imports = false;
+      if (!parser_.opts.one_file) {
+        tracked_imported_namespaces_.clear();
```

Comment:
I recommend put this part in a reset function.

Reply:
Done.